sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -292,43 +296,66 @@ def call(self):
     side_input_values = [self._side_input_values[side_input]
                          for side_input in self._applied_ptransform.side_inputs]
 
-    try:
-      evaluator = self._transform_evaluator_registry.get_evaluator(
-          self._applied_ptransform, self._input_bundle,
-          side_input_values, scoped_metrics_container)
-
-      if self._fired_timers:
-        for timer_firing in self._fired_timers:
-          evaluator.process_timer_wrapper(timer_firing)
-
-      if self._input_bundle:
-        for value in self._input_bundle.get_elements_iterable():
-          evaluator.process_element(value)
-
-      with scoped_metrics_container:
-        result = evaluator.finish_bundle()
-        result.logical_metric_updates = metrics_container.get_cumulative()
+    # Switch to turn on/off the retry of bundles.
+    pipeline_options = self._evaluation_context.pipeline_options
+    if not pipeline_options.view_as(DirectOptions).direct_runner_bundle_retry:
+      TransformExecutor._MAX_RETRY_PER_BUNDLE = 1
```

Comment:
<!--new_thread; commit:3dff6d2c80d8bc66e25f0ee7b6e369a34b938862; resolved:0-->
Can you move this to `__init__` and rename this `self._max_retries`?  The reasoning being that we shouldn't globally modify the class property of `TransformExecutor`.